#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 213c4bc1eb1f45695cc3955cc722ebb363dcbdd0 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Tue, 26 Mar 2019 19:22:44 +0100
Subject: [PATCH] core,w32,glib: Fix build of w32-glib-io.c

* src/w32-glib-io.c (_gpgme_io_pipe, _gpgme_io_connect): Do not
use TRACE_SUC in a return statement.
---
 src/w32-glib-io.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/src/w32-glib-io.c b/src/w32-glib-io.c
index 8c8722ce..e2e3b8ab 100644
--- a/src/w32-glib-io.c
+++ b/src/w32-glib-io.c
@@ -421,12 +421,13 @@ _gpgme_io_pipe (int filedes[2], int inherit_idx)
       return TRACE_SYSRES (-1);
     }

-  return TRACE_SUC ("read=0x%x/%p, write=0x%x/%p, channel=%p",
-		     filedes[0],
-		     (HANDLE) _get_osfhandle (giochannel_table[filedes[0]].fd),
-		     filedes[1],
-		     (HANDLE) _get_osfhandle (giochannel_table[filedes[1]].fd),
-		     giochannel_table[1 - inherit_idx].chan);
+  TRACE_SUC ("read=0x%x/%p, write=0x%x/%p, channel=%p",
+	     filedes[0],
+	     (HANDLE) _get_osfhandle (giochannel_table[filedes[0]].fd),
+	     filedes[1],
+	     (HANDLE) _get_osfhandle (giochannel_table[filedes[1]].fd),
+	     giochannel_table[1 - inherit_idx].chan);
+  return 0;
 }


@@ -1081,5 +1082,7 @@ _gpgme_io_connect (int fd, struct sockaddr *addr, int addrlen)
       return TRACE_SYSRES (-1);
     }

-  return TRACE_SUC ("");
+  TRACE_SUC ("");
+
+  return 0;
 }
--
2.20.1
